services/authorize/client.go

```diff
@@ -207,6 +205,20 @@ func (c *client) DialUsingCredentialsWithContext(ctx context.Context, sess *sess
 	return err
 }
 
+func reconnectRetryInterceptor(newClientConn reconnect.NewConnectionFunc) grpc.DialOption {
+	retryIC := grpc_retry.UnaryClientInterceptor(
+		grpc_retry.WithBackoff(grpc_retry.BackoffLinear(100*time.Millisecond)),
+		grpc_retry.WithCodes(codes.Unavailable, codes.ResourceExhausted, codes.Aborted),
+	)
+
+	reconnectIC := reconnect.UnaryInterceptor(
+		reconnect.WithNewConnection(newClientConn),
+	)
+
+	dialOptsReconnectRetry := grpc.WithChainUnaryInterceptor(reconnectIC, retryIC) // first one is outer, being called last
```

Comment:
No need to create a variable, return it directly.

Reply:
Code gets so messy, rather not